// Protobuf Version 3.5.1

syntax = "proto3";

message Vec3
{
    float X = 1;
    float Y = 2;
    float Z = 3;
}

message PCheckRequest
{
    int32 Core = 1;
    int32 Spp = 2;
    int32 Width = 3;
    int32 Height = 4;
}

message PSceneRequest
{
    message Triangle
    {
        Vec3 A = 1;
        Vec3 B = 2;
        Vec3 C = 3;
    }

    message Material
    {
        string Name = 1;
        int32 Type = 2;
        repeated float Parameters = 3;
    }

    message Mesh
    {
        string MatName = 1;
        repeated Triangle Triangles = 2;
    }

    message Sphere
    {
        string MatName = 1;
        Vec3 Center = 2;
        float Radious = 3;
    }

    repeated Material Materials = 1;
    repeated Mesh Meshs = 2;
    repeated Sphere Spheres = 3;

    Vec3 CameraLocation = 4;
    Vec3 CameraForward = 5;
    Vec3 CameraRight = 6;
}

message PRenderPixelResponse
{
    int32 X = 1;
    int32 Y = 2;
    Vec3 Result = 3;
}